### DISCUSSION TITLE
useConditionalEffect callback is called even if conditions unchanged

### DISCUSSION THREAD

**Discussion by nikolay-borzov** [react-hookz/web#156] [General]
Looking at `useConditionalEffect`  I'd expect it would behave as `useEffect` that has condition inside callback:

```js
useConditionalEffect(() => {
  // Act
}, [dep1, dep2])

// Above is the same as

useEffect(() => {
  if(dep1 && dep2) {
    // Act
  }
}, [dep1, dep2])
```

But instead the callback called on next re-render even if `dep1` and `dep2` are unchanged.

Question: Is this behavior by design?

**Comment by xobotyi** [comment]
@nikolay-borzov the design is descccribed in [docs](https://react-hookz.github.io/web/?path=/docs/lifecycle-useconditionaleffect--example)=)

> Like `useEffect` but callback invoked only if conditions match predicate. By default, predicate matches if all conditions are truthy.

So by default, whenever condition `dep1 && dep2` is truthy - effect will be executed.

**Reply by xobotyi** [reply]
I'm planning to add parrameter, that will change the behaviour to the one you expect, smth like 4th argument for the hook.

Create the issue please if you okay with such solution.

**Reply by nikolay-borzov** [reply]
Sure, I'll create an issue
